groups/bsl/bslstl/bslstl_optional.t.cpp

```diff
@@ -758,120 +980,11218 @@ struct UsesAllocatorArgT<MyClass2a> : bsl::true_type {
+
+    if (verbose)
+        printf("\nTESTING RELATIONAL OPERATORS"
+               "\n============================\n");
+    if (veryVerbose)
+        printf("\tComparison with an 'optional'.\n");
+    {
+        testCase31_imp_a<bsl::optional<int>, bsl::optional<MyClass2> >();
+
+        testCase31_imp_a<bsl::optional<MyClass2>, bsl::optional<int> >();
+
+#ifdef BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
+        testCase31_imp_a<std::optional<int>, bsl::optional<int> >();
+
+        testCase31_imp_a<bsl::optional<int>, std::optional<int> >();
+
+        testCase31_imp_a<std::optional<MyClass2>, bsl::optional<MyClass2> >();
+
+        testCase31_imp_a<bsl::optional<MyClass2>, std::optional<MyClass2> >();
+
+        testCase31_imp_a<std::optional<int>, bsl::optional<MyClass2> >();
+
+        testCase31_imp_a<bsl::optional<MyClass2>, std::optional<int> >();
+
+        testCase31_imp_a<std::optional<MyClass2>, bsl::optional<int> >();
+
+        testCase31_imp_a<bsl::optional<int>, std::optional<MyClass2> >();
+#endif  //BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
+    }
+    if (veryVerbose)
+        printf("\tComparison with a non 'optional' .\n");
+    {
+        testCase31_imp_b<bsl::optional<int>, MyClass2>();
+
+        testCase31_imp_b<bsl::optional<MyClass2>, int>();
+    }
+    if (veryVerbose)
+        printf("\tComparison with a nullopt_t .\n");
+    {
+        testCase31_imp_c<int>();
+        testCase31_imp_c<MyClass2>();
+    }
+}
+
+template <class TYPE>
+void TestDriver<TYPE>::testCase30()
+{
+    // --------------------------------------------------------------------
+    // TESTING ALLOCATOR EXTENDED 'make_optional' FACILITY
+    //
+    //  In this test, we are ensuring that the 'optional' created using
+    //  allocator extended 'make_optional' facility is created using the given
+    //  arguments and without unnecessary copies.  We are not worried about the
+    //  allocator policy of the allocator-aware type, as choosing the allocator
+    //  policy when creating an 'optional' is tested in constructor tests.
+    //
+    //  This test verifies the invocation of allocator extended 'make_optional'
+    //  where the type is deduced.
+    //
+    //
+    // Concerns:
+    //: 1 Invoking allocator extended 'make_optional' creates an 'optional'
+    //:   with the value of the argument.
+    //: 2 In a single non-allocator argument of allocator extended
+    //:   'make_optional' invocation, the value type of the optional is deduced
+    //:   from the non-allocator argument.
+    //: 3 Argument is perfectly forwarded.
+    //: 4 The specified allocator is used as the allocator for the returned
+    //:   'optional' object.
+    //: 5 No unnecessary objects of 'value_type' are created.
+    //
+    //
+    // Plan:
+    //: 1 Call allocator extended 'make_optional' to create an 'optional'.  As
+    //:   an argument use an lvalue of the desired 'value_type'.  Check
```

Comment:
```suggestion
    //:   an argument use an lvalue of the desired 'value_type'.  Verify that
```

Reply:
done

Reply:
replaced all instances of "check" with "verify"